### PR TITLE
Added the profile editing functionality.

### PR DESCRIPTION
**Pull request by ymaheshwari1**
# Related Issues or bug
  - Account Holder should be able to edit his/her profile details.

Fixes: #25

# Proposed Changes
 - Added a button on the profile page to update the User's details
 - Created a new page where the user can edit his/her details.

# Screenshots
ezgif com-video-to-gif

### CODE REVIEW

**Comment by harsh-9in** [comment]
@Kajol-Kumari  Plz review this :)

File: mysite/users/forms.py [hunk 1]
```diff
@@ -13,4 +13,11 @@ class Meta:
...
-        fields = ['image']
\ No newline at end of file
+        fields = ['image']
+
+class UserProfileUpdateForm(UserChangeForm):
+    password = ReadOnlyPasswordHashField(label="Password", widget=forms.HiddenInput(),
+                                                help_text="Password Field Hidden")
+    class Meta:
+        model = User
+        fields = ['username', 'email']
```
Comment (Kajol-Kumari) [inline mysite/users/forms.py:23]: Please provide an extra blank line here.

File: mysite/users/templates/user/editProfile.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      <legend class="border-bottom mb-4 ">Update Information</legend>
+      {{ form | crispy }}
+    </fieldset>
+    <div class="form-group">
+      <button type="submit" class="btn btn-primary">Update</button>
+    </div>
+  </form>
+</div>
+
+{% endblock %}
```
Comment (Kajol-Kumari) [inline mysite/users/templates/user/editProfile.html:19]: same here. please prove an extra blank line at the end.

File: mysite/users/views.py [hunk 3]
```diff
@@ -56,4 +56,16 @@ def profileUpdate(request, pk):
...
+def editProfile(request):
+    if request.method == 'POST':
+        # server = get_object_or_404(Profile, pk=pk)
+        form = UserProfileUpdateForm(request.POST, instance=request.user)
+        if form.is_valid():
+            form.save()
+            messages.success(request,f'Your profile information updated successfully')
+            return redirect(profile)
+    form = UserProfileUpdateForm(instance = request.user)
+    return render(request, 'user/editProfile.html', {'form': form})
```
Comment (Kajol-Kumari) [inline mysite/users/views.py:71]: same here.

File: mysite/users/views.py [hunk 4]
```diff
@@ -56,4 +56,16 @@ def profileUpdate(request, pk):
...
+    return render(request,'user/profile.html',{'form': form})
+
+@login_required
+def editProfile(request):
+    if request.method == 'POST':
+        # server = get_object_or_404(Profile, pk=pk)
+        form = UserProfileUpdateForm(request.POST, instance=request.user)
+        if form.is_valid():
+            form.save()
+            messages.success(request,f'Your profile information updated successfully')
```
Comment (Kajol-Kumari) [inline mysite/users/views.py:68]: If I am not wrong, there is a typo in the success message, please correct it.

File: mysite/users/views.py [hunk 5]
```diff
@@ -56,4 +56,16 @@ def profileUpdate(request, pk):
...
             return redirect('profile')
     form = UserProfileForm()
-    return render(request,'user/profile.html',{'form': form})
\ No newline at end of file
+    return render(request,'user/profile.html',{'form': form})
+
+@login_required
+def editProfile(request):
+    if request.method == 'POST':
+        # server = get_object_or_404(Profile, pk=pk)
```
Comment (Kajol-Kumari) [inline mysite/users/views.py:64]: Please remove the commented lines unless it's some info regarding the functionlaity.

File: mysite/users/views.py (hunk 4 above)
Comment (ymaheshwari1) [reply mysite/users/views.py:68]: @Kajol-Kumari If you are talking about the "successfully" then it is correct.

Comment (Kajol-Kumari) [reply mysite/users/views.py:68]: No, I am taking about the `,f'`

Comment (ymaheshwari1) [reply mysite/users/views.py:68]: Here ,f denotes formatted string. If we use any variables/expressions then it will be replaced by its value

Comment (Kajol-Kumari) [reply mysite/users/views.py:68]: yeah, thanks for letting me know :)

Comment (ymaheshwari1) [reply mysite/users/views.py:68]: @Kajol-Kumari I have added the GIF. Please have a look

**Pull request merged**